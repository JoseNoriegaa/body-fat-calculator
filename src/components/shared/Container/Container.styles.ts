// External dependencies
import { createUseStyles } from 'react-jss';

// Internal dependencies
import breakpoints from '@/utilities/breakpoints';

const useStyles = createUseStyles({
  container: {
    margin: '0 auto',
    maxWidth: breakpoints.get('xl'),
    padding: '0 24px',
  },
});

export default useStyles;
