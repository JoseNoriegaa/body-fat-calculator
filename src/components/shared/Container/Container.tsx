// External dependencies
import React from 'react';
import clsx from 'clsx';

// Internal dependencies
import useStyles from './Container.styles';

// Types & Interfaces
interface IContainerProps {
  className?: string;
}

function Container({ children, className }: React.PropsWithChildren<IContainerProps>) {
  const classes = useStyles();

  return <div className={clsx(classes.container, className)}>{children}</div>;
}

export default Container;
