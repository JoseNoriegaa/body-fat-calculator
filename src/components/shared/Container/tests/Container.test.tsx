// External dependencies
import { render, screen } from '@testing-library/react';

// Internal dependencies
import Container from '../Container';

describe('<Container />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Container>
        <span>children</span>
      </Container>
    );

    expect(screen.getByText('children')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
