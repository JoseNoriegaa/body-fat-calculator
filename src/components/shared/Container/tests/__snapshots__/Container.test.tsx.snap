// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-0-2-1"
  >
    <span>
      children
    </span>
  </div>
</DocumentFragment>
`;
